components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/UserSessionStore.java

```diff
@@ -74,17 +74,22 @@ public static UserSessionStore getInstance() {
     public void storeUserData(String userId, String userName, int tenantId, String userDomain, int idPId)
             throws UserSessionException {
 
-        try (Connection connection = IdentityDatabaseUtil.getDBConnection();
-             PreparedStatement preparedStatement = connection
-                     .prepareStatement(SQLQueries.SQL_INSERT_USER_STORE_OPERATION)) {
-            preparedStatement.setString(1, userId);
-            preparedStatement.setString(2, userName);
-            preparedStatement.setInt(3, tenantId);
-            preparedStatement.setString(4, (userDomain == null) ? FEDERATED_USER_DOMAIN : userDomain.toUpperCase());
-            preparedStatement.setInt(5, idPId);
-            preparedStatement.executeUpdate();
-            if (!connection.getAutoCommit()) {
-                connection.commit();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            try ( PreparedStatement preparedStatement = connection
+                    .prepareStatement(SQLQueries.SQL_INSERT_USER_STORE_OPERATION)) {
+                preparedStatement.setString(1, userId);
+                preparedStatement.setString(2, userName);
+                preparedStatement.setInt(3, tenantId);
+                preparedStatement.setString(4, (userDomain == null) ? FEDERATED_USER_DOMAIN :
+                        userDomain.toUpperCase());
+                preparedStatement.setInt(5, idPId);
+                preparedStatement.executeUpdate();
+                if (!connection.getAutoCommit()) {
```

Comment:
checking if (!connection.getAutoCommit()) {
will not help here. Because  connection.setAutocommit(false) is a begin of transaction.
We need to keep track of transactions done in IdentityDatabaseUtil.getDBConnection())
and close it with IdentityDatabaseUtil.commit(); and IdentityDatabaseUtil.rollback()